File: backend/src/services/vectorservice.js
import { qdrant } from "../config/qdrant.js";

export async function ensureCollection() {
  const collections = await qdrant.getCollections();

  const exists = collections.collections.some(
    c => c.name === process.env.QDRANT_COLLECTION
  );

  if (!exists) {
    await qdrant.createCollection(process.env.QDRANT_COLLECTION, {
      vectors: {
        size: 3072,        // ✅ Gemini embedding dimension
        distance: "Cosine"
      }
    });

    console.log("Qdrant collection created:", process.env.QDRANT_COLLECTION);
  } else {
    console.log("Qdrant collection already exists");
  }
}

File: backend/src/services/vector_service.py
from qdrant_client.models import Distance, FieldCondition, Filter, MatchValue, VectorParams

from src.core.config import get_settings
from src.integrations.qdrant import get_qdrant


def ensure_collection() -> None:
    settings = get_settings()
    qdrant = get_qdrant()
    collections = qdrant.get_collections()
    exists = any(collection.name == settings.qdrant_collection for collection in collections.collections)

    if not exists:
        qdrant.create_collection(
            collection_name=settings.qdrant_collection,
            vectors_config=VectorParams(size=3072, distance=Distance.COSINE),
        )
        print("Qdrant collection created:", settings.qdrant_collection)
    else:
        print("Qdrant collection already exists")


def delete_document_vectors(*, document_id: str, user_id: str) -> None:
    settings = get_settings()
    qdrant = get_qdrant()
    collections = qdrant.get_collections()
    exists = any(collection.name == settings.qdrant_collection for collection in collections.collections)

    if not exists:
        return

    qdrant.delete(
        collection_name=settings.qdrant_collection,
        points_selector=Filter(
            must=[
                FieldCondition(key="documentId", match=MatchValue(value=document_id)),
                FieldCondition(key="userId", match=MatchValue(value=user_id)),
            ]
        ),
        wait=True,
    )
